R/pwr_2p.R

```diff
@@ -1,748 +1,1048 @@
-      # d <- ifelse(calc == "es", r$adp, lst$adp)
-      # d <- round(d, 2)
-      # power <- ifelse(calc == "power", r$power, lst$pow)
-      # alpha <- ifelse(calc == "alpha", r$alpha, lst$alpha)
-      # alt <- lst$alt
-      #
-      # n_text <- ifelse(n1 == n2,
-      #                  gettextf("a sample size of %s in each group", n1),
-      #                  gettextf("group sample sizes of %s and %s, respectively", n1, n2)
-      # )
-      #
-      # if (alt == "two.sided") {
-      #   tail_text <- gettext("two-sided")
-      #   null_text <- "<i>\u03B4=</i>0,"
-      #   alt_text <- "|<i>\u03B4</i>|<i>\u2265</i>"
-      #   crit_text <- gettext("criteria")
-      # } else {
-      #   tail_text <- gettext("one-sided")
-      #   null_text <- "<i>\u03B4\u2264</i>0,"
-      #   alt_text <- "<i>\u03B4\u2265</i"
-      #   crit_text <- gettext("criterion")
-      # }
-      #
-      # str <- paste(
-      #   "<p>",
-      #   gettextf("The figure above shows two sampling distributions: the sampling distribution of the <i>estimated</i> effect size when <i>%s=</i>0 (left), and when <i>%s=</i>%s (right).", "\u03B4", "\u03B4", d),
-      #   gettextf("Both assume %s.", n_text),
-      #   "</p><p>",
-      #   gettextf("The vertical dashed lines show the %s we would set for a %s test with <i>α=</i>%s.", crit_text, tail_text, alpha),
-      #   gettextf("When the observed effect size is far enough away from 0 to be more extreme than the %s we say we 'reject' the null hypothesis.", crit_text),
-      #   gettextf("If the null hypothesis were true and %s the evidence would lead us to wrongly reject the null hypothesis at most %s%% of the time.", null_text, 100 * alpha),
-      #   "</p><p>",
-      #   gettextf("On the other hand, if <i>%s%s</i>%s, the evidence would exceed the criterion  &mdash; and hence we would correctly claim that <i>%s%s</i>0 &mdash; at least %s%% of the time.", "\u03B4", "\u2265", d, "\u03B4", "\u2265", 100 * round(power, 3)),
-      #   gettextf("The design's power for detecting effects of %s%s is thus %s.", alt_text, d, round(power, 3)),
-      #   "</p>"
-      # )
-      #
-      #
-      # html[["text"]] <- str
+.runTest2P <- function(jaspResults, options) {
+  stats <- .prepareStats(options)
+
+  ## Compute results
+  results <- try(.computeTest2P(jaspResults, options, stats))
+  if (inherits(results, "try-error")) {
+    .quitAnalysis(gettext("Unable to compute the power results. Try to enter less extreme values for the input parameters."))
+  }
+
+  .initPowerTabTest2P(jaspResults, options, results, stats)
+
+  if (options$text) {
+    .initPowerESTabTest2P(jaspResults, options, results, stats)
+  }
+
+  ## Populate tables and plots
+  if (options$text) {
+    .populateIntro(jaspResults, options)
+  }
+
+  if (options$powerContour) {
+    .preparePowerContourTest2P(jaspResults, options, results, stats)
+    if (options$text) {
+      .populateContourTextTest2P(jaspResults, options, results, stats)
```

Comment:
These parts are also a bit repetetive...

For example, instead of having in each analysis

```r
if(options[["text"]])
  .populateIntro(jaspResults, options)
```

You could just define `.populateIntro` with
```
.populateIntro <- function(jaspResults, options) {
  if(!options[["text"]]) return()

  # rest of the original function
}
```

Then you don't have to do the check every time you call this function and as a result the analysis functions become a bit more readable.

Reply:
Ah, yeah, this is the part where I struggled a bit to find a nice way of porting this over from the object oriented programming style (which solves this more elegantly).

Will update this for any common functions, I think for the analysis specific functions it's probably more readable this way? And it would be duplicated either way 🤔